storage/zynq7000-flash/flashdrv.c

```diff
@@ -718,13 +680,24 @@ int flashdrv_devInit(storage_t *strg)
 	strg->dev->blk = malloc(sizeof(storage_blk_t));
 	if (strg->dev->blk == NULL) {
 		resourceDestroy(reg->lock);
-		free(reg->buff);
 		free(strg->dev);
+		free(strg->dev->ctx);
 		free(strg->dev->mtd);
 		return -ENOMEM;
 	}
 	strg->dev->blk->ops = &blkOps;
 
+	cacheOps.ctx->id = id;
+	reg->cache = cache_init(strg->start + strg->size, BLK_CACHE_SECNUM * secSz, secSz, &cacheOps);
```

Comment:
Why strg->start?
```suggestion
	reg->cache = cache_init(strg->size, BLK_CACHE_SECNUM * secSz, secSz, &cacheOps);
```

Reply:
cache do not care about starting offsets, it should be handled in callbacks (according to info in devCtx)